--- slang-1.4.5/src/sltermin.c.orig	Sun Feb 10 08:39:20 2002
+++ slang-1.4.5/src/sltermin.c	Wed Feb 27 02:27:57 2002
@@ -105,7 +105,31 @@
     * I will also look into the use of setreuid, seteuid and setregid, setegid.
     * FIXME: Priority=medium
     */
+    /* If your system lacks setfsuid/getfsuid either write
+       equivalent support or dont use slang to build setuid/setgid
+       apps like Mutt */
+ 
+   if(setfsuid(getuid())==-1)
+   {
+       perror("setfsuid");
+       return NULL;
+   }
+   if(setfsgid(getgid())==-1)
+   {
+       perror("setfsgid");
+       return NULL;
+   }
    fp = fopen (file, "rb");
+   if(setfsuid(geteuid())==-1)
+   {
+       perror("setfsuid");
+       return NULL;
+   }
+   if(setfsgid(getegid())==-1)
+   {
+       perror("setfsgid");
+       return NULL;
+   }
    if (fp == NULL) return NULL;
 
    if ((12 == fread ((char *) buf, 1, 12, fp) && (MAGIC == make_integer (buf))))
@@ -285,7 +309,7 @@
 
 	     if (sizeof (file) > strlen (tidir) + 2 + strlen (term))
 	       {
-		  sprintf (file, "%s/%c/%s", tidir, *term, term);
+		  snprintf (file, sizeof (file), "%s/%c/%s", tidir, *term, term);
 		  if (NULL != (fp = open_terminfo (file, ti)))
 		    break;
 	       }
