--- slang-2.1.2/modules/Makefile.in.orig	2007-09-08 04:47:29.000000000 +0200
+++ slang-2.1.2/modules/Makefile.in	2007-09-11 08:24:31.836264453 +0200
@@ -79,27 +79,27 @@
 	cp ../src/config.h .
 
 slsmg-module.so: slsmg-module.c config.h
-	$(CC_SHARED) $(INCS) slsmg-module.c -o slsmg-module.so $(LIBS)
+	$(CC_SHARED) $(LDFLAGS) $(INCS) slsmg-module.c -o slsmg-module.so $(LIBS)
 newt-module.so: newt-module.c config.h
-	$(CC_SHARED) $(INCS) newt-module.c -o newt-module.so -lnewt $(LIBS)
+	$(CC_SHARED) $(LDFLAGS) $(INCS) newt-module.c -o newt-module.so -lnewt $(LIBS)
 termios-module.so: termios-module.c config.h
-	$(CC_SHARED) $(INCS) termios-module.c -o termios-module.so $(LIBS)
+	$(CC_SHARED) $(LDFLAGS) $(INCS) termios-module.c -o termios-module.so $(LIBS)
 select-module.so: select-module.c config.h
-	$(CC_SHARED) $(INCS) select-module.c -o select-module.so $(LIBS)
+	$(CC_SHARED) $(LDFLAGS) $(INCS) select-module.c -o select-module.so $(LIBS)
 fcntl-module.so: fcntl-module.c config.h
-	$(CC_SHARED) $(INCS) fcntl-module.c -o fcntl-module.so $(LIBS)
+	$(CC_SHARED) $(LDFLAGS) $(INCS) fcntl-module.c -o fcntl-module.so $(LIBS)
 varray-module.so: varray-module.c config.h
-	$(CC_SHARED) $(INCS) varray-module.c -o varray-module.so $(LIBS)
+	$(CC_SHARED) $(LDFLAGS) $(INCS) varray-module.c -o varray-module.so $(LIBS)
 pcre-module.so: pcre-module.c config.h
-	$(CC_SHARED) $(INCS) $(PCRE_INC) pcre-module.c -o pcre-module.so $(PCRE_LIB) $(LIBS)
+	$(CC_SHARED) $(LDFLAGS) $(INCS) $(PCRE_INC) pcre-module.c -o pcre-module.so $(PCRE_LIB) $(LIBS)
 onig-module.so: onig-module.c config.h
-	$(CC_SHARED) $(INCS) $(ONIG_INC) onig-module.c -o onig-module.so $(ONIG_LIB) $(LIBS)
+	$(CC_SHARED) $(LDFLAGS) $(INCS) $(ONIG_INC) onig-module.c -o onig-module.so $(ONIG_LIB) $(LIBS)
 png-module.so: png-module.c config.h
-	$(CC_SHARED) $(INCS) $(PNG_INC) png-module.c -o png-module.so $(PNG_LIB) $(LIBS)
+	$(CC_SHARED) $(LDFLAGS) $(INCS) $(PNG_INC) png-module.c -o png-module.so $(PNG_LIB) $(LIBS)
 socket-module.so: socket-module.c config.h
-	$(CC_SHARED) $(INCS) socket-module.c -o socket-module.so $(SOCKET_LIBS) $(LIBS)
+	$(CC_SHARED) $(LDFLAGS) $(INCS) socket-module.c -o socket-module.so $(SOCKET_LIBS) $(LIBS)
 iconv-module.so: iconv-module.c config.h
-	$(CC_SHARED) $(INCS) $(ICONV_INC) iconv-module.c -o iconv-module.so $(ICONV_LIB) $(LIBS)
+	$(CC_SHARED) $(LDFLAGS) $(INCS) $(ICONV_INC) iconv-module.c -o iconv-module.so $(ICONV_LIB) $(LIBS)
 
 # Note that Windows/Cygwin modules need execute permissions
 install: all
