diff -ur slang-1.4.4/slsh/Makefile.in slang-1.4.4.my/slsh/Makefile.in
--- slang-1.4.4/slsh/Makefile.in	Wed Feb 21 02:17:31 2001
+++ slang-1.4.4.my/slsh/Makefile.in	Sun Jun 24 13:33:22 2001
@@ -6,8 +6,8 @@
 #---------------------------------------------------------------------------
 prefix 		= @prefix@
 exec_prefix 	= @exec_prefix@
-SLANG_INC	= -I@includedir@
-SLANG_LIB	= -L@libdir@ -lslang
+SLANG_INC	= -I@includedir@ -I../src
+SLANG_LIB	= -L@libdir@ -lslang -L../src/elfobjs
 #---------------------------------------------------------------------------
 DL_LIB		= @DYNAMIC_LINK_LIB@
 
